src/test/ui/unsafe/rfc-2585-unsafe_op_in_unsafe_fn.rs

```diff
@@ -0,0 +1,58 @@
+#![feature(unsafe_block_in_unsafe_fn)]
+#![warn(unsafe_op_in_unsafe_fn)]
```

Comment:
I think this should be `deny`, not warn, so that we get errors. I believe the test framework *permits* one to specify where a warning is expected, but doesn't make the test fail if there are add'l warnings beyond those that were listed.

Reply:
I changed it according to RalfJung's review:
> Please make this `warn`, to make sure that is sufficient to trigger the "unused_unsafe" lint.

I think the solution is to have `deny` at the top level, but have an additional function where the lint level is `warn`, just like we have a function where the lint level is `allow`? If warnings can't make tests fail, that won't catch a missing warning, though. Maybe another solution is to have both `#[warn(unsafe_op_in_unsafe_fn)]` and `#[deny(warnings)]`?

Reply:
@nikomatsakis isn't it the case that when you have `// WARN` anywhere in the test, then it insists that you specify *all* warnings or else the test fails?